### PR TITLE
Loading indicator fix for plots

### LINKED ISSUE #5219: Plots are sometimes not showing (or too eagerly hiding) spinners
Spun out from https://github.com/nasa/openmct/issues/4777

#### Summary
The issue here is that when historical data retrieval is extremely slow there is no visual indication provided in the plot that historical requests are still outstanding. ie. there is no spinner being shown.

In attempting to reproduce this, the plot did eventually fill in the missing data, it's just taking several minutes for Yamcs to return historical data at times.

Historically we overlaid a spinner until the data was returned, so restoring that functionality is a fine solution in the short term.

Longer term though, this is not the right approach. A spinner is very intrusive, and real-time data may be available first and may be obscured by the spinner. Something more akin to the progress bar use in tables is probably a better bet.

#### Steps to Reproduce
1. Navigate to a plot of data (real data, will not work with SWGs)
2. Navigate away
3. Open the network tab
4. Throttle the connection to slow 3G
5. Navigate back to the plot. Observe that the plot shows no spinner at all.

#### Impact Check List

- [X] Data loss or misrepresented data?
- [X] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [X] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?

### PR DESCRIPTION
**Pull request by alizenguyen**
Closes #5219 

### Describe your changes:

- Fixed bug where a loading indicator was not displaying when retrieving historical data. 
- Change loading indicator from spinner to progress bar so plot is not blocked when loading historical data.

Attached is a recording of what it should look like. 

Fix for stacked plots are awaiting changes and will be in a separate pull request.

https://user-images.githubusercontent.com/29556173/170150557-4b12d0ff-2c0f-4c32-bf59-a3e45cc68133.mov

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by alizenguyen** [comment]
Just something to note for now! There should be a loading indicator for each individual plot in stacked plots when waiting for data. Switching to progress bars, we might want to find some way to differentiate which progress bar goes to which plot better. Below is a screen recording of what it looks like.

https://user-images.githubusercontent.com/29556173/170322306-af2593e8-d6c9-4c3c-926a-9ca63150666b.mov

**Comment by unlikelyzero** [comment]
@shefalijoshi we've added a visual test for this area, however we're not able to tick the clock in such a way that mimic a delayed response from a telemetry object. Is there any way you could work with @alizenguyen to figure out how to mock that in the visual test?

**Comment by alizenguyen** [comment]
@unlikelyzero Hi! Sorry for the delay, wasn't feeling too good. Here is a video of how to add a loading delay time!

https://user-images.githubusercontent.com/29556173/171871413-5b713bca-6381-43e5-93e5-22c8252017a4.mov

**Review by unlikelyzero — APPROVED** [review]

**Comment by alizenguyen** [comment]
I saw that the new stacked plots changes got merged, so I included the loading fix for stack plots in this PR as well

File: src/plugins/plot/MctPlot.vue [hunk 1]
```diff
@@ -484,7 +484,7 @@ export default {
                     end: range.max,
                     domain: this.config.xAxis.get('key')
                 })
-                    .then(this.stopLoading());
+                    .then(this.stopLoading.bind(this));
```
Comment (shefalijoshi) [inline src/plugins/plot/MctPlot.vue:487]: Hmm. Don't we want to call `stopLoading` here?

Comment (alizenguyen) [reply src/plugins/plot/MctPlot.vue:487]: It was calling it right away which was stopping the loading too soon before the data was being loading. Binding it to this fixed the issue. I followed the example that was set by loadSeriesData above.

File: example/generator/generatorWorker.js [hunk 2]
```diff
@@ -133,17 +134,19 @@
...
-            data: request.spectra ? {
-                wavelength: data.map((item) => {
-                    return item.wavelength;
-                }),
-                cos: data.map((item) => {
-                    return item.cos;
-                })
-            } : data
-        });
+        setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (alizenguyen) [reply example/generator/generatorWorker.js:137]: loadDelay a prop and defaults to 0, so it should always return defined. Do you mean a check if the loadDelay is 0?

File: src/plugins/plot/Plot.vue [hunk 3]
```diff
@@ -75,6 +77,13 @@ export default {
             status: ''
         };
     },
+    computed: {
+        progressLoad() {
+            return {
+                progressPerc: undefined
```
[the remark this answers is not in the document]
Comment (shefalijoshi) [reply src/plugins/plot/Plot.vue:83]: Can you talk to @jvigliotta ? And see what the best solution is? Thanks

**Comment by unlikelyzero** [comment]
@shefalijoshi we're ready for re-review

**Comment by shefalijoshi** [comment]
@alizenguyen This needs to be merged into release/2.0.5 right? Can you change the target branch and make sure nothing from master snuck in?

**Comment by alizenguyen** [comment]
@shefalijoshi  Ah, think it may be easier for me to create a new branch from release/2.0.5 and cherry pick the commits. I'll create a new pull request.

**Pull request closed without merging**